Automattic/jetpack
# Publicize: Improve primary form component.

### Branched off from https://github.com/Automattic/jetpack/pull/21085

This PR reimplements the form of the Publicize plugin.
Currently, it has two `form` and `form-unwrapped` components that make it kind of confusing.
With these changes, the form in a single component, in a single file. Also, it adds two React hooks making the whole implementation more readable, at least from my PoV 😅 

Fixes https://github.com/Automattic/jetpack/issues/21086

#### Changes proposed in this Pull Request:

Publicize: Improve primary form.

#### Jetpack product discussion



#### Does this pull request change what data or activity we track or use?



#### Testing instructions:






* It doesn't change the functionality of the feature, so the first testing is confirm that everything is working as expected:
  - create a new post
  - add some content
  - start to edit the message to share the post on the social media
  - publish the post
  - check the post is shared in the social media account 

**Testing the hooks in the Publicize component**
You can take a look at the hooks in the `<PublicizeForm />` component using the React dev tool:

<img width="1235" alt="Screen Shot 2021-09-14 at 3 50 02 PM" src="https://user-images.githubusercontent.com/77539/133316695-950a8d21-cefa-4eb4-a31b-a0fc89fb0743.png">


**Testing the data propagation**
Similarly, you can take a look at the `core` sub store tree, using the Redux dev tool.

<img width="1247" alt="Screen Shot 2021-09-14 at 3 46 10 PM" src="https://user-images.githubusercontent.com/77539/133316329-63700675-f8ad-4076-876f-f02d1ba1b141.png">

The image above shows how the data is stored in the subtree while the user edits the publicize content via the message box component.
